main.js

```diff
@@ -1,46 +1,92 @@
-const { app, BrowserWindow, Tray, Menu } = require('electron');
-const path = require('path');
-
-let tray = null;
-let mainWindow = null;
-
-app.whenReady().then(() => {
-    // Create system tray icon
-    tray = new Tray(path.join(__dirname, 'icon.png')); // Add a custom icon
-    const contextMenu = Menu.buildFromTemplate([
-        { label: 'Open ElectroBlocks', click: () => createWindow() },
-        { type: 'separator' },
-        { label: 'Quit', click: () => app.quit() }
-    ]);
-    tray.setToolTip('ElectroBlocks Tray App');
-    tray.setContextMenu(contextMenu);
-
-    // Create window when clicking tray icon
-    tray.on('click', () => {
-        if (mainWindow) {
-            mainWindow.show();
-        } else {
-            createWindow();
-        }
-    });
-});
+import { app, BrowserWindow, Menu, dialog, shell } from "electron";
+import express from "express";
+import { SerialPort } from "serialport";
+
+let mainWindow;
+
+// Start Express Server Inside Electron
+const startExpressServer = () => {
+  const expressApp = express();
+  const PORT = 4000;
+
+  expressApp.get("/ports", async (req, res) => {
+    try {
+      const ports = await SerialPort.list();
+      res.json(ports.map((port) => ({
+        path: port.path,
+        manufacturer: port.manufacturer || "Unknown",
+        vendorId: port.vendorId || "N/A",
+        productId: port.productId || "N/A",
+        serialNumber: port.serialNumber || "N/A"
+      })));
+    } catch (error) {
+      res.status(500).json({ error: error.message });
+    }
+  });
+
+  expressApp.listen(PORT, () => {
+    console.log(`Server running at http://localhost:${PORT}/ports`);
+  });
+};
+
+// Function to Get Available Ports
+const getAvailablePorts = async () => {
+  try {
+    const ports = await SerialPort.list();
+    return ports;
+  } catch (error) {
+    return [];
+  }
+};
+
+// Create Electron Window
+app.whenReady().then(async () => {
+  mainWindow = new BrowserWindow({

```

Comment:
We want to continue to use a tray app instead of windows.
![Screenshot 2025-03-20 at 8 21 27 AM](https://github.com/user-attachments/assets/d01d4ed7-f97b-471e-b1ad-774dbc70cb1c)